cli_tools/gce_image_publish/publish/publish.go

```diff
@@ -585,3 +624,48 @@ func calculateExpiryDate(deleteAfter string) (*time.Time, error) {
 
 	return &expiryDate, nil
 }
+
+func createRollOut(zones []*compute.Zone, rolloutStartTime time.Time, rolloutRate int) *computeAlpha.RolloutPolicy {
+	rp := computeAlpha.RolloutPolicy{}
+	rp.DefaultRolloutTime = rolloutStartTime.Format(time.RFC3339)
+
+	var regions map[string][]string
+	regions = make(map[string][]string)
+	maxRegionLength := 0
+
+	// Build a map of all the regions and determine the max number of zones in a region.
+	for _, z := range zones {
+		regions[z.Region] = append(regions[z.Region], z.Name)
+		if len(regions[z.Region]) > maxRegionLength {
+			maxRegionLength = len(regions[z.Region])
+		}
+	}
+
+	// Order the list of zones in each region.
+	for _, value := range regions {
+		sort.Strings(value)
+	}
+
+	// zoneList is the ordered list of zones to apply the rollout policy to.
+	var zoneList []string
+
+	// zoneList's order should be the first zone from each region, then second zone from each region, third zone from each region, etc.
+	// us-central1-a, us-central2-b, us-central3-c, us-central1-a, us-central2-b, us-central3-c
+	for zoneCount := 0; zoneCount < maxRegionLength; zoneCount++ {
+		for _, value := range regions {
```

Comment:
minor: `value` -> `regionZones` or similar

Reply:
Thanks the more descriptive variable name helps.